HW0/P4/P4.py

```diff
@@ -51,12 +63,29 @@
     # A = ?
     # a = ?
     # s = ?
-
+	
     # Initial conditions for s0
+    sk_list = [[0],[0],[2],[15],[3.5],[4.0]] # first column is initial condition
+    sk = np.array([[0],[0],[2],[15],[3.5],[4.0]])
+    A = np.array([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],
```

Comment:
Instead of doing hard coding,  you could also make use of sparse numpy array, more specifically using np.diag() is more memory efficient.